frontend/src/metabase/visualizations/components/LegendVertical.jsx

```diff
@@ -65,41 +65,43 @@ export default class LegendVertical extends Component {
     }
     return (
       <ol className={cx(className, styles.Legend, styles.vertical)}>
-        {items.map((title, index) => (
-          <li
-            key={index}
-            ref={"item" + index}
-            className="flex flex-no-shrink"
-            onMouseEnter={e =>
-              onHoverChange &&
-              onHoverChange({
-                index,
-                element: ReactDOM.findDOMNode(this.refs["legendItem" + index]),
-              })
-            }
-            onMouseLeave={e => onHoverChange && onHoverChange()}
-          >
-            <LegendItem
-              ref={"legendItem" + index}
-              title={Array.isArray(title) ? title[0] : title}
-              color={colors[index % colors.length]}
-              isMuted={
-                hovered && hovered.index != null && index !== hovered.index
+        {items.map((title, index) => {
+          const isMuted =
+            hovered && hovered.index != null && index !== hovered.index;
+          return (
+            <li
+              key={index}
+              ref={"item" + index}
+              className="flex flex-no-shrink"
+              onMouseEnter={e =>
+                onHoverChange &&
+                onHoverChange({
+                  index,
+                  element: ReactDOM.findDOMNode(
+                    this.refs["legendItem" + index],
+                  ),
+                })
               }
-              showTooltip={false}
-            />
-            {Array.isArray(title) && (
-              <span
-                className={cx("LegendItem", "flex-align-right pl1", {
-                  muted:
-                    hovered && hovered.index != null && index !== hovered.index,
-                })}
-              >
-                {title[1]}
-              </span>
-            )}
-          </li>
-        ))}
+              onMouseLeave={e => onHoverChange && onHoverChange()}
+            >
+              <LegendItem
+                ref={"legendItem" + index}
+                title={Array.isArray(title) ? title[0] : title}
+                color={colors[index % colors.length]}
+                isMuted={isMuted}
+                showTooltip={false}
+              />
+              {Array.isArray(title) && (
+                <span
```

Comment:
Could be a good opportunity to make a styled component that accepts a muted prop, rather than apply style directly to the element, assuming there are no static viz impacts. 
cc @alxnddr

Reply:
Normally I would do this but here we are using some class names. Unfortunately `@emotion/styled` does not have a `.attrs` method like `styled-components` does, so we wouldn't be able to pass a `className` prop into the styled component ([example](https://medium.com/@justiceotuya/how-to-add-classnames-to-styled-components-f1f88518ab51)).

I would prefer not to define the styles in a styled component and then add the class names in the JSX, I think it's better to keep all the style definitions in the same file.

Reply:
Can we move the styles from those classes into a styled component as well? Or do we have tests that depend on `.LegendItem`?